engine/src/cubos/engine/gizmos/plugin.cpp

```diff
@@ -193,9 +216,49 @@ static void drawGizmosSystem(Write<Gizmos> gizmos, Write<GizmosRenderer> gizmosR
     iterateGizmos(gizmos->screenGizmos, screenInfo, *gizmosRenderer, *deltaTime);
 }
 
+static void processInput(Write<GizmosRenderer> gizmosRenderer, Write<Gizmos> gizmos,
+                         EventReader<WindowEvent> windowEvent)
+{
+    for (const auto& event : windowEvent)
+    {
+        if (std::holds_alternative<MouseMoveEvent>(event))
+        {
+            gizmosRenderer->lastMousePosition = std::get<MouseMoveEvent>(event).position;
+        }
+        else if (std::holds_alternative<MouseButtonEvent>(event))
+        {
+            if (std::get<MouseButtonEvent>(event).button == cubos::core::io::MouseButton::Left)
+            {
+                gizmosRenderer->mousePressed = std::get<MouseButtonEvent>(event).pressed;
+            }
+        }
+        else if (std::holds_alternative<ResizeEvent>(event))
+        {
+            gizmosRenderer->resizeTexture(std::get<ResizeEvent>(event).size);
+        }
+    }
+
+    auto* texBuffer = new uint16_t[(std::size_t)gizmosRenderer->textureSize.x *
+                                   (std::size_t)gizmosRenderer->textureSize.y * (std::size_t)2U];
+
+    gizmosRenderer->idTexture->data(texBuffer);
```

Comment:
OpenGL provides [glReadPixels](https://registry.khronos.org/OpenGL-Refpages/gl4/html/glReadPixels.xhtml) which seems a much better fit to use here. I think that with that you wouldn't even need the integer masking down there, as you could directly extract the data in the format you desire. I think using it instead is a good idea - copying the whole screen texture every frame just to check a single pixel is a big cost to pay - and the engine is already slow as it is :cry: